src/components/Button/src/Button.vue

```diff
@@ -17,6 +17,19 @@
 		v-bind="$attrs"
 		v-on="$listeners"
 	>
+		<div
+			v-if="variant === 'success' || 'error'"
+			:class="$s.IconAligner"
+		>
+			<component
+				:is="iconComponent"
+				:class="[
+					$s.Icon,
+					$s[`type_${type}`],
+				]"
+				inline
+			/>
+		</div>
```

Comment:
i would prefer to add this into the MainText section somehow so that it mimics
```
<m-button>
  <check-circle class="icon" />
</m-button>
```
but i couldn't figure it out.  when i placed it inside the MainText section, the svg colors weren't passed in correctly.

Reply:
A couple things here:

1. I think this outer wrapper should be a `<span>` so it's natively an inline element inside the button, unless you need it to be a `<div>` for some reason.
2. We should be able to use the existing icon prefix slot perhaps to do this, so we don't need to create another condition where an icon is added. Unless that messes up the preferred alignment of this type of button, as the prefix icons are always aligned left? cc @pretzelhammer for your thoughts on this

Reply:
@ashjtan 

When you tried

```vue
<m-button>
    <check-circle class="icon" /> Some text here
</m-button>
```

Did you have these styles on the `.icon` class?:

```css
.icon {
    width: 16px;
    height: 16px;
    fill: currentColor;
    // and maybe even
    stroke: currentColor;
}
```

It should be inheriting the text color if it has those styles.

@mattmewton 

MInputs and MSelects have icon slots because it's otherwise not possible to set icons inside them. With buttons is possible to pass icons straight through using the default slot so I'd rather not add unnecessary icon slots to button.

Reply:
@pretzelhammer to clarify, ur example
```
<m-button>
    <check-circle class="icon" /> Some text here
</m-button>
```
works fine. but the idea was to have the icons included as part of the variant automatically so that i could get the same result by just doing `<m-button variant="success">`. i wanted to achieve this by adding in the icon in the MainText section like
```
<span :class="$s.MainText">
    <span v-if="isSuccessOrErrorVariant>
        <check-circle or alert-circle class="icon" />
    </span>
    <!-- @slot Button label -->
    <slot />
</span>
```
but when i did it this way, even with the `.icon` styles you put above, the svg renders w/ `fill=none` so the icon is basically invisible. that's why i ended up adding all the complexity i did, with the icons [outside of the MainText section](https://github.com/square/maker/pull/50/files#diff-a63a381a9ca0c90e9b1158a50b1c4bfbb0cb75b95488098dba716e13324c2342R20-R32).

given this, do you think that the icon should not be included in the maker `ActionBarButton` (i.e. what you advocated for in a [below comment](https://github.com/square/maker/pull/50#discussion_r604170952))?